tests/e2e/specs/shopper/checkout-payments.test.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import {
+	merchant,
+	setCheckbox,
+	settingsPageSaveChanges,
+	verifyCheckboxIsSet,
+} from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getNormalPagePermalink,
+	shopper,
+	visitPostOfType,
+} from '../../../utils';
+
+const block = {
+	name: 'Checkout',
+};
+
+const simpleProductName = 'Woo Single #1';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
+
+describe( `${ block.name } Block (frontend)`, () => {
+	let productPermalink;
+
+	beforeAll( async () => {
+		// prevent CartCheckoutCompatibilityNotice from appearing
+		await page.evaluate( () => {
+			localStorage.setItem(
+				'wc-blocks_dismissed_compatibility_notices',
+				'["checkout"]'
+			);
+		} );
+		await merchant.login();
+
+		// Get product page permalink.
+		await visitPostOfType( simpleProductName, 'product' );
+		productPermalink = await getNormalPagePermalink();
+
+		await merchant.logout();
+	} );
+
+	afterAll( async () => {
+		// empty cart from shortcode page
+		await shopper.goToCart();
+		await shopper.removeFromCart( simpleProductName );
+		await page.evaluate( () => {
+			localStorage.removeItem(
+				'wc-blocks_dismissed_compatibility_notices'
+			);
+		} );
+	} );
+
+	it( 'allows customer to choose available payment methods', async () => {
+		await shopper.goToCheckoutBlock();
+		await page.goto( productPermalink );
+		await shopper.addToCart();
+		await shopper.goToCheckoutBlock();
+
+		await expect( page ).toClick(
```

Comment:
I wonder if we shouldn't ensure that the correct payment method gets selected after clicking on it. The following code only ensures that an element with the class `wc-block-components-payment-method-label` and inner text `Direct bank transfer` gets clicked, but we should follow up with a code to test if it did get indeed selected. Sometimes, an element gets clicked without getting the proper behavior, and the test will still pass.

```js
await expect( page ).toClick(
	'.wc-block-components-payment-method-label',
	{
		text: 'Direct bank transfer',
	}
);
```

For example, I have tested the following code and the test passed! I clicked on an H1 element:

```js
await expect( page ).toClick( '.entry-title', {
	text: 'Checkout Block',
} );
```

Reply:
Great suggestion! I updated the test in a way that I check if the payment option had been checked after it had been selected.